extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/runtime/src/main/java/io/quarkus/resteasy/reactive/jackson/runtime/serialisers/JacksonMessageBodyWriter.java

```diff
@@ -81,23 +93,69 @@ public void writeResponse(Object o, Type genericType, ServerRequestContext conte
             // First test the names to see if JsonView is used. We do this to avoid doing reflection for the common case
             // where JsonView is not used
             ResteasyReactiveResourceInfo resourceInfo = context.getResteasyReactiveResourceInfo();
-            if ((resourceInfo != null) && resourceInfo.getMethodAnnotationNames().contains(JSON_VIEW_NAME)) {
-                Method method = resourceInfo.getMethod();
-                if (handleJsonView(method.getAnnotation(JsonView.class), o, stream)) {
-                    return;
+            if (resourceInfo != null) {
+                Set<String> methodAnnotationNames = resourceInfo.getMethodAnnotationNames();
+                if (methodAnnotationNames.contains(CUSTOM_SERIALIZATION)) {
+                    Method method = resourceInfo.getMethod();
+                    if (handleCustomSerialization(method, o, genericType, stream)) {
+                        return;
+                    }
+                } else if (methodAnnotationNames.contains(JSON_VIEW_NAME)) {
+                    Method method = resourceInfo.getMethod();
+                    if (handleJsonView(method.getAnnotation(JsonView.class), o, stream)) {
+                        return;
+                    }
                 }
             }
-            writer.writeValue(stream, o);
+            defaultWriter.writeValue(stream, o);
         }
         // we don't use try-with-resources because that results in writing to the http output without the exception mapping coming into play
         stream.close();
     }
 
+    // TODO: this can definitely be made faster if necessary by optimizing the use of the map and also by moving the creation of the
+    //  biFunction to build time
```

Comment:
Yeah, I thought this would be build-time

Reply:
We can move it build time, but given that it's only a limited number of invocation of the constructor (basically one per usage of the annotation), I think it's fine to leave it runtime for the time being.

Reply:
Yes, agreed, that's not user-visible.